extension.ts

```diff
@@ -112,6 +112,15 @@ export async function activateInternal(ctx: vscode.ExtensionContext, perfStats:
     this.properties.isActivationEvent = 'true';
     this.measurements.mainFileLoad = (perfStats.loadEndTime - perfStats.loadStartTime) / 1000;
 
+    const extension = vscode.extensions.getExtension('PeterJausovec.vscode-docker');
+    if (extension) {
+      // temporary workaround for https://github.com/microsoft/vscode-docker/issues/1008
+      // We will essentially defer all functionality to the old publisher's extension if we detect both installed
+      vscode.commands.executeCommand('setContext', 'isOldPublisherInstalled', true);
+      this.properties.isOldPublisherInstalled = 'true';
+      return;
+    }
```

Comment:
With this PR, the extension would hide all commands when it recognizes the old extension. I don't see code at the registerCommand sites to ensure the command is not run. 
What happens when a dependent extension calls say `docker.deployToWebApp` before the extension is activated? Does : 
1. The command hit both extensions? Then this code will make sure the commands are hidden, but the command might still run, while also running the older extension copy
2. It pings this extension, which prevents the other extension from running the command.

Reply:
The key part is the `return` line. This extension will never even register the commands if we detect the old publisher.

Reply:
Should we alert them and ask to uninstall the old one?  Otherwise how will they ever see new functionality?

Reply:
The problem is that if they uninstall the old one, they will immediately be recommended to install it again if they open a Dockerfile. So yes - we need to do that, but not until VS Code stable ships with the updated recommendation id

I think it makes sense to include that logic in our next actual release. Currently this release and the old release have the exact same functionality. I want to be able to say "uninstall the old one _so that you can get this new functionality_"